### PR TITLE
Angular Inputs for Storybook

### LINKED ISSUE #1848: Add Inputs Component - Angular NDS
**AC:**
#2227 
#2230 
#2231

### PR DESCRIPTION
**Pull request by RVMendoza**
## What does this PR do?
Add angular input stories
- Added percentage Input on react as well

### Associated Issue 
Fixes #1848, #2232 

### Documentation
 - [x] Update Spark Docs React
 - [x] Update Spark Docs Angular

### Code
 - [x] Build Component in Spark Angular
 - [x] Unit Testing in Spark Angular with `gulp test-angular` (100% coverage, 100% passing)
 - [x] Unit Testing in Spark React with `gulp test-react` (100% coverage, 100% passing)

### Accessibility
- [x] New changes abide by [accessibility requirements](https://sparkdesignsystem.com/docs/accessibility)

### Browser Testing (current version and 1 prior)
  - [x] Google Chrome
  - [x] Google Chrome (Mobile)
  - [x] Mozilla Firefox
  - [x] Mozilla Firefox (Mobile)
  - [x] Microsoft Internet Explorer 11 (only this specific version of IE)
  - [x] Microsoft Edge
  - [x] Apple Safari
  - [x] Apple Safari (Mobile)

### CODE REVIEW

File: angular/projects/spark-angular/src/lib/components/inputs/sprk-input.stories.ts [hunk 1]
```diff
@@ -0,0 +1,649 @@
...
+import { SprkSelectionLabelModule } from '../../directives/inputs/sprk-selection-label/sprk-selection-label.module';
+import { SprkTextareaContainerModule } from './sprk-textarea-container/sprk-textarea-container.module';
+import { SprkFieldErrorModule } from '../../directives/inputs/sprk-field-error/sprk-field-error.module';
+import { SprkFormatterPhoneNumberModule } from '../../directives/inputs/formatters/sprk-formatter-phone-number/sprk-formatter-phone-number.module';
+import { SprkFormatterSsnModule } from '../../directives/inputs/formatters/sprk-formatter-ssn/sprk-formatter-ssn.module';
+import { SprkFormatterDateModule } from '../../directives/inputs/formatters/sprk-formatter-date/sprk-formatter-date.module';
+import { SprkFormatterMonetaryModule } from '../../directives/inputs/formatters/sprk-formatter-monetary/sprk-formatter-monetary.module';
+import { SprkHelperTextModule } from '../../directives/inputs/sprk-helper-text/sprk-helper-text.module';
+import { SprkIconInputContainerModule } from './sprk-icon-input-container/sprk-icon-input-container.module';
+import { SprkDatepickerModule } from '../../directives/inputs/sprk-datepicker/sprk-datepicker.module';
```
Comment (guern1kn) [inline angular/projects/spark-angular/src/lib/components/inputs/sprk-input.stories.ts]: Some of these exceed our line length max.

File: angular/projects/spark-angular/src/lib/components/inputs/sprk-input.stories.ts [hunk 2]
```diff
@@ -0,0 +1,649 @@
...
+      <sprk-selection-item-container>
+        <input
+          type="checkbox"
+          name="checkbox_input"
+          [(ngModel)]="checkbox_input2"
+          sprkSelectionInput
+          #checkboxInput2="ngModel"
+          data-id="checkbox-2"
+          id="checkbox-2"
+           (change)="onSelect($event)"
```
Comment (guern1kn) [inline angular/projects/spark-angular/src/lib/components/inputs/sprk-input.stories.ts]: Extra space here

File: angular/projects/spark-angular/src/lib/components/inputs/sprk-input.stories.ts [hunk 3]
```diff
@@ -0,0 +1,649 @@
...
+      <sprk-selection-item-container>
+        <input
+        type="checkbox"
+        name="checkbox_input"
+        [(ngModel)]="checkbox_input3"
+        sprkSelectionInput
+        #checkboxInput3="ngModel"
+        data-id="checkbox-3"
+        id="checkbox-3"
+        (change)="onSelect($event)"
```
Comment (guern1kn) [inline angular/projects/spark-angular/src/lib/components/inputs/sprk-input.stories.ts]: Indent one more

File: angular/projects/spark-angular/src/lib/components/inputs/sprk-input.stories.ts [hunk 4]
```diff
@@ -0,0 +1,649 @@
...
+        <optgroup label="Grouped Options">
+          <option value="g1">Grouped Option 1</option>
+          <option value="g2">Grouped Option 2</option>
+          <option value="g3">Grouped Option 3</option>
+        </optgroup>
+      </select>
+      <sprk-icon
+      iconType="chevron-down"
+      additionalClasses="sprk-c-Icon--stroke-current-color sprk-b-SelectContainer__icon"
+      sprk-select-icon
```
Comment (guern1kn) [inline angular/projects/spark-angular/src/lib/components/inputs/sprk-input.stories.ts]: Indent 1 more

File: angular/projects/spark-angular/src/lib/components/inputs/sprk-input.stories.ts [hunk 5]
```diff
@@ -0,0 +1,649 @@
...
+  template: `
+    <sprk-textarea-container>
+      <label for="textarea-1" sprkLabel>Description</label>
+      <textarea
+      name="textarea_input"
+      id="textarea-1"
+      [(ngModel)]="textarea_input"
+      #textareaInput="ngModel"
+      data-id="textarea-1"
+      sprkInput
```
Comment (guern1kn) [inline angular/projects/spark-angular/src/lib/components/inputs/sprk-input.stories.ts]: Indent 1 more

File: angular/projects/spark-angular/src/lib/components/inputs/sprk-input.stories.ts [hunk 6]
```diff
@@ -0,0 +1,649 @@
...
+          sprkFieldError
+        >
+          <sprk-icon
+            iconType="exclamation-filled-small"
+            additionalClasses="sprk-b-ErrorIcon"
+          >
+          </sprk-icon>
+          <div class="sprk-b-ErrorText">Invalid SSN.</div>
+        </div>
+    </sprk-input-container>
```
Comment (guern1kn) [inline angular/projects/spark-angular/src/lib/components/inputs/sprk-input.stories.ts]: Indented 1 too many

File: angular/projects/spark-angular/src/lib/components/inputs/sprk-input.stories.ts [hunk 7]
```diff
@@ -0,0 +1,649 @@
...
+        <div
+          [hidden]="ssnInput.valid || ssnInput.pristine"
+          sprkFieldError
+        >
+          <sprk-icon
+            iconType="exclamation-filled-small"
+            additionalClasses="sprk-b-ErrorIcon"
+          >
+          </sprk-icon>
+          <div class="sprk-b-ErrorText">Invalid SSN.</div>
```
Comment (guern1kn) [inline angular/projects/spark-angular/src/lib/components/inputs/sprk-input.stories.ts]: Should there be a period here?

File: angular/projects/spark-angular/src/lib/components/inputs/sprk-input.stories.ts [hunk 8]
```diff
@@ -0,0 +1,649 @@
...
+  template: `
+    <sprk-input-container>
+      <label sprkLabel>Search Input</label>
+      <input
+      name="search_input"
+      type="search"
+      [(ngModel)]="search_input"
+      #searchInput="ngModel"
+      data-id="search-1"
+      sprkInput
```
Comment (guern1kn) [inline angular/projects/spark-angular/src/lib/components/inputs/sprk-input.stories.ts]: Indent 1 more

File: angular/projects/spark-angular/src/lib/components/inputs/sprk-input.stories.ts [hunk 9]
```diff
@@ -0,0 +1,649 @@
...
+        [(ngModel)]="inline_search_input"
+        #inlineSearchInput="ngModel"
+        data-id="input-search-1"
+        sprkInput
+      >
+    </sprk-icon-input-container>
+  `,
+});
+
+
```
Comment (guern1kn) [inline angular/projects/spark-angular/src/lib/components/inputs/sprk-input.stories.ts]: Extra line here

File: angular/projects/spark-angular/src/lib/components/inputs/sprk-input.stories.ts [hunk 10]
```diff
@@ -0,0 +1,649 @@
...
+export const monetaryInput = () => ({
+  moduleMetadata: modules,
+  template: `
+    <sprk-icon-input-container
+      iconContainerClasses="sprk-b-TextInputIconContainer--has-text-icon"
+    >
+      <label class="sprk-b-Label--monetary" sprkLabel> Payment </label>
```
Comment (guern1kn) [inline angular/projects/spark-angular/src/lib/components/inputs/sprk-input.stories.ts]: Extra spaces around "Payment"

File: angular/projects/spark-angular/src/lib/components/inputs/sprk-input.stories.ts [hunk 11]
```diff
@@ -0,0 +1,649 @@
...
+      />
+      <div
+        [hidden]="monetaryInput.valid || monetaryInput.pristine"
+        sprkFieldError
+      >
+        <sprk-icon
+          iconType="exclamation-filled-small"
+          additionalClasses="sprk-b-ErrorIcon"
+        ></sprk-icon>
+        <div class="sprk-b-ErrorText">Invalid amount.</div>
```
Comment (guern1kn) [inline angular/projects/spark-angular/src/lib/components/inputs/sprk-input.stories.ts]: Depending on where we land with the period at the end of the SSN error, let's make sure this one matches.

File: angular/projects/spark-angular/src/lib/components/inputs/sprk-input.stories.ts [hunk 12]
```diff
@@ -0,0 +1,649 @@
...
+        class="sprk-b-InputContainer__input--has-icon-right"
+        name="percentage"
+        type="tel"
+        sprkInput
+      />
+    </sprk-icon-input-container>
+  `,
+});
+
+
```
Comment (guern1kn) [inline angular/projects/spark-angular/src/lib/components/inputs/sprk-input.stories.ts]: Extra line here

**Comment by afebbraro** [comment]
@RVMendoza, this one can't be merged until those conflicts from yesterday are fixed

**Pull request merged**